lib/expand.js

```diff
@@ -225,6 +227,102 @@ class Expander {
     }
   }
 
+  expandHierarchy(element) {
+    const setCurrentElementAsOriginalModifier = (value) => {
+      if (value.options) {
+        value.options.forEach(v => setCurrentElementAsOriginalModifier(v));
+      } else {
+        value.constraints.forEach(c => c.lastModifiedBy = element.identifier);
+      }
+    }
+
+    if (element.basedOn.length == 0) {
+      if (element.value) setCurrentElementAsOriginalModifier(element.value);
+      if (element.fields) element.fields.forEach(f => setCurrentElementAsOriginalModifier(f));
+
+    } else {
+      for (const baseID of element.basedOn) {
+        if (baseID instanceof models.TBD) continue;
+
+        const base = this.lookup(baseID);
+        if (typeof base === 'undefined') {
+          logger.error('Reference to non-existing base: %s. ERROR_CODE:12002', baseID.fqn);
+          continue;
+        }
+
+      
+        if (base.hierarchy.length == 0 && base.basedOn.length > 0) {
+          this.expandHierarchy(base);
+        }
+
+        var manageValueInheritance = (parent, child) => {
+          if (child.inheritedFrom == null) {
+            if (parent.inheritedFrom) {
+              child.inheritedFrom = parent.inheritedFrom;
+            } else {
+              child.inheritedFrom = base.identifier;
+            };
+          }
+
+          if (child.constraints.length > 0 && child.identifier) {
```

Comment:
Why the check on `child.identifier`?  You don't seem to use `child.identifier` below, so it's not a defensive check (as best I can tell).  One thing this check _does_ do is it blocks out `ChoiceValue`s (since they don't have an `identifier`).  Is that intentional?

Reply:
This was my original intention. I believe my code was originally written in a way that required the matching of the child.identifier to parent.identifier, although, looking at it now, there is no longer any reason to single them out as I've since improved the check to not require that.